### PR TITLE
Add support for dotty 0.27.0-RC1 and update mdoc

### PR DESCRIPTION
**Pull request by tgodzik**
The updated mdoc now uses `toString` and inbuilt mechanism for reporting types, which solves the issue with pprint dependency.

Fixes https://github.com/scalameta/metals/issues/2024

### CODE REVIEW

File: tests/cross/src/test/scala/tests/pc/CompletionSuite.scala [hunk 1]
```diff
@@ -460,7 +460,7 @@ class CompletionSuite extends BaseCompletionSuite {
   )
 
   check(
-    "commit1",
+    "commit1".tag(IgnoreScalaVersion("0.27.0-RC1")),
```
Comment (tgodzik) [inline tests/cross/src/test/scala/tests/pc/CompletionSuite.scala:463]: Not sure why, but in 0.27.0-RC1 completions stopped working from things in the standard library.

Comment (tgodzik) [reply tests/cross/src/test/scala/tests/pc/CompletionSuite.scala:463]: I will follow up in dotty and ask.

File: build.sbt [hunk 2]
```diff
@@ -201,8 +202,8 @@ lazy val V = new {
   def scala2Versions = nonDeprecatedScala2Versions ++ deprecatedScala2Versions
 
   // Scala 3
-  def nonDeprecatedScala3Versions = Seq(scala3, "0.25.0")
-  def deprecatedScala3Versions = Seq("0.25.0-RC2", "0.26.0-RC1")
+  def nonDeprecatedScala3Versions = Seq(scala3, scala3Candidate)
```
Comment (tgodzik) [inline build.sbt:205]: It seems that we have a large regression in 0.27 and no completions coming from scala library are not displayed.
This way most large tests will be run on 0.26, which works and I will focus on fixing it upstream before the release.

File: tests/unit/src/main/scala/tests/QuickBuild.scala [hunk 3]
```diff
@@ -117,7 +117,9 @@ case class QuickBuild(
           s"org.scala-lang:scala-library:$scalaVersion",
           s"org.scala-lang:scala-reflect:$scalaVersion"
         )
-      } ++ libraryDependencies
```
Comment (tgodzik) [inline tests/unit/src/main/scala/tests/QuickBuild.scala:120]: Fun fact, this never added anything for Scala 3, which is why some tests worked even with bad dependencies. CrossReferenceLspSuite and RenameCrossSuite were the only ones that were problematic.

File: build.sbt (hunk 2 above)
Comment (tgodzik) [reply build.sbt:205]: Link to the regression https://github.com/lampepfl/dotty/issues/9760

**Review by romanowski — APPROVED** [review]

File: tests/unit/src/main/scala/tests/QuickBuild.scala (hunk 3 above)
Comment (tgodzik) [reply tests/unit/src/main/scala/tests/QuickBuild.scala:120]: Ach! This is due to the fact that else was operating on the whole expression with the `++`

**Comment by tgodzik** [comment]
This is blocking for @odersky so I will go ahead and merge. If anyone has any suggestions you can report them here, I will try to fix them later in a separate PR.

File: tests/unit/src/main/scala/tests/QuickBuild.scala (hunk 3 above)
Comment (ckipp01) [reply tests/unit/src/main/scala/tests/QuickBuild.scala:120]: 👀 good catch ha

**Pull request merged**